shared/configHelper.js

```diff
@@ -115,18 +115,50 @@ function printVersion(cli) {
 function printArgs(cli, commandName) {
     getArgsExpanded(cli, commandName)
         .filter(arg => !arg.hidden)
-        .forEach(arg => logInfo('    ' + (!arg.required  ? '[' : '') + '--' + arg.name + '=' + arg.description + (!arg.required ? ']' : ''), COLOR.magenta));
+        .forEach(arg => logInfo('    ' + (!arg.required ? '[' : '') + '--' + arg.name + '=' + arg.description + (!arg.required ? ']' : ''), COLOR.magenta));
 }
 
-function listTemplates(cli) {
+function listTemplates(cli, commandLineArgs) {
     var cliName = cli.name;
-    var applicableTemplates = getTemplates(cli);
 
-    logInfo('\nAvailable templates:\n', COLOR.cyan);
-    for (var i=0; i<applicableTemplates.length; i++) {
+    // Parse command line arguments to extract templatesource or templaterepouri
+    var templateSource = null;
+    var templateRepoUri = null;
+    if (commandLineArgs && commandLineArgs.length > 0) {
+        try {
+            var argsMap = commandLineUtils.parseArgs(commandLineArgs);
+            templateSource = argsMap[SDK.args.templateSource && SDK.args.templateSource.name];
+            templateRepoUri = argsMap[SDK.args.templateRepoUri.name];
+        } catch (error) {
+            // If argument parsing fails, continue without templateRepoUri
+        }
+    }
+
+    var source = templateSource || templateRepoUri;
+    var applicableTemplates = getTemplates(cli, source);
+
+    // Show which template repository is being used
+    if (source) {
+        logInfo('\nAvailable templates from custom repository:\n', COLOR.cyan);
+        logInfo('Repository: ' + source, COLOR.cyan);
+    } else {
+        logInfo('\nAvailable templates:\n', COLOR.cyan);
+    }
+
+    for (var i = 0; i < applicableTemplates.length; i++) {
         var template = applicableTemplates[i];
-        logInfo((i+1) + ') ' + template.description, COLOR.cyan);
-        logInfo(cliName + ' ' + SDK.commands.createwithtemplate.name + ' --' + SDK.args.templateRepoUri.name + '=' + template.path, COLOR.magenta);
+        logInfo((i + 1) + ') ' + template.description, COLOR.cyan);
+        // If using custom repository, include it in the command
+        var templateUri = template.path;
+        if (source) {
+            // Parse the repository URI to separate repo and branch
```

Comment:
There is a helper function for that: see https://github.com/forcedotcom/SalesforceMobileSDK-Package/blob/dev/shared/utils.js#L331